src/main/java/edu/umich/its/cpm/EmailFormatter.java

```diff
@@ -0,0 +1,161 @@
+        ArrayList<String> prunedHeaders = new ArrayList<String>();
+        for (String header : getHeaders()) {
+            if (!StringUtils.containsIgnoreCase(header, "content-type")) {
+                prunedHeaders.add(header);
+            }
+        }
+        return prunedHeaders;
+    }
+
+    public HashMap<String, ArrayList<String>> getAttachments() {
+        HashMap<String, ArrayList<String>> attachmentMap = new HashMap<String, ArrayList<String>>();
+        ArrayList<HashMap<String, Object>> attachments = (ArrayList<HashMap<String, Object>>) messageMap.get("attachments");
+        for (HashMap<String, Object> attachment : attachments) {
+            final String mimeType = (String) attachment.get("type");
+            final String url = (String) attachment.get("url");
+            final String name = (String) attachment.get("name");
+            attachmentMap.put((String) attachment.get("id"), new ArrayList<String>() {{
+                add(mimeType);
+                add(name);
+                add(url);
+            }});
+        }
+        return attachmentMap;
+
+    }
+
+
+    public String rawEmail() {
+        Session session = null;
+        MimeMessage message = new MimeMessage(session);
+        String rawEmail = null;
+        try {
+            Multipart multipart = new MimeMultipart();
+
+            //Body of the email
+            BodyPart msgBodyPart = new MimeBodyPart();
+            String body = getBody();
+            if(!body.isEmpty()) {
+                msgBodyPart.setContent(body, "text/plain; charset=UTF-8");
+                multipart.addBodyPart(msgBodyPart);
+            }
+
+            // attachment part of email
+            HashMap<String, ArrayList<String>> attachments = getAttachments();
+            if (!attachments.isEmpty()) {
+                Set<String> attachmentIDs = attachments.keySet();
+                for (String id : attachmentIDs) {
+                    msgBodyPart = new MimeBodyPart();
+                    ArrayList<String> attachmentMetaData = attachments.get(id); // [Mimetype, fileName, attachmentUrl]
+                    String attachmentUrl = attachmentMetaData.get(2);
+                    String mimeType = attachmentMetaData.get(0);
+                    String fileName = attachmentMetaData.get(1);
+                    byte[] fileContent = attachmentHandler.getAttachmentContent(attachmentUrl, request);
+                    msgBodyPart.setDataHandler(new DataHandler(new ByteArrayDataSource(fileContent, mimeType)));
+                    msgBodyPart.setFileName(fileName);
+                    msgBodyPart.addHeader("Content-Transfer-Encoding", "base64");
+                    multipart.addBodyPart(msgBodyPart);
+
+                }
+
+            }
+            message.setContent(multipart);
+            ByteArrayOutputStream output = new ByteArrayOutputStream();
+            message.writeTo(output);
+            rawEmail = output.toString();
```

Comment:
The emailText that i am returning here is from java mail client and usually they follow the RFC822( which is the basic documentation) and many more standards, Mime standard for attachments extending the RFC822. So basically i don't need to be very specific in naming that. i have added comments specifying this. The thing is if i rename it as`emailRFC822Text` standard then when my `EmailFormatter.java` class have more formatting class like MBOX, PDF. In all those case this `emailText` will act like a base ingredient to create the Mbox or pdf format of email. so that why i don't want to rename the variable